File: src/main/java/com/example/sorokov/entity/UserEntity.java
package com.example.sorokov.entity;
import jakarta.persistence.*;
import lombok.*;
import lombok.experimental.Accessors;
import org.hibernate.annotations.SourceType;
import org.hibernate.annotations.UpdateTimestamp;
import org.springframework.data.annotation.CreatedDate;

import java.util.Date;

@Entity
@Builder
@Table(name = "\"user\"")
@Data
@AllArgsConstructor
@NoArgsConstructor
@Accessors(chain = true)
public class UserEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String surname;
    private String name;
    @CreatedDate
    @Column(name = "created_at", nullable = false)
    private Date createdAt;
    @Column(unique = true, nullable = false)
    private String email;
    @Column(name = "updated_at", nullable = false)
    @UpdateTimestamp(source = SourceType.DB)
    private Date updatedAt;
    @Column(name = "avatar_path")
    private String avatarPath;
    private String password;

}

File: SorokChatServer.Infrastructure/Entities/UserEntity.cs
using SorokChatServer.Infrastructure.ValueObjects;

namespace SorokChatServer.Infrastructure.Entities;

public class UserEntity : BaseEntity
{
    public UserEntity(long id, DateTime createdAt, DateTime updatedAt, Name name, Email email, Password password)
        : base(id, createdAt, updatedAt)
    {
        Name = name;
        Email = email;
        Password = password;
    }

    public UserEntity()
    {
    }

    public Name Name { get; set; } = null!;

    public Email Email { get; set; } = null!;

    public Password Password { get; set; } = null!;
}